Game/Designers/BuffDesigner/BuffDesigner.cpp

```diff
@@ -43,24 +44,24 @@ BuffBlueprint& BuffDesigner::DesignBuff(SBuffRequirements aBuffRequirements)
 	return *myBlueprints.insert(std::move(blueprint));
 }
 
-EBuffType BuffDesigner::DecideBuffType() const
+EBuffType BuffDesigner::DecideBuffType()
 {
-	return SystemPtr<Randomizer>()->SelectOne<EBuffType, EBuffType::Debuff, EBuffType::Buff>();
+	return Get<Randomizer>().SelectOne<EBuffType, EBuffType::Debuff, EBuffType::Buff>();
 }
 
-f32 BuffDesigner::DecideBuffStrength() const
+f32 BuffDesigner::DecideBuffStrength()
 {
-	return SystemPtr<Randomizer>()->GetRandomNumber(50.f, 150.f);
+	return Get<Randomizer>().GetRandomNumber(50.f, 150.f);
 }
 
-f32 BuffDesigner::DecideBuffIntensity() const
+f32 BuffDesigner::DecideBuffIntensity()
 {
-	return std::powf(SystemPtr<Randomizer>()->GetRandomNumber(1.f, 1000.f) / 1000.f, 2.f) * 10.f;
+	return std::powf(Get<Randomizer>().GetRandomNumber(1.f, 1000.f) / 1000.f, 2.f) * 10.f;
 }
 
-void BuffDesigner::MakeBuffBetter(const f32 aHowMuch, SBuffDesign& aBuffDesign) const
+void BuffDesigner::MakeBuffBetter(const f32 aHowMuch, SBuffDesign& aBuffDesign)
 {
-	switch (SystemPtr<Randomizer>()->GetRandomNumber<i32>(0, 1))
+	switch (Get<Randomizer>().GetRandomNumber<i32>(0, 1))
```

Comment:
Might be worth thinking about having GetRandomNumber be const and making the seed mutable; getting a random number _feels_ like a pure operation